### DISCUSSION TITLE
Using individual components in external repositories using a private npm registry

### DISCUSSION THREAD

**Discussion by c0d3x** [storybookjs/storybook#26983] [Help]
### Summary

I want to be able to maintain the components individually by version and deploy them to private NPM registry of the company.
Is there a Storybook way for this to combine development with the use of Chromatic and other features?

So my components packages will look like this:
```
"@MyCompany/button":"0.0.1",
"@MyCompany/input":"0.2.1",
```

- Then major changes to components will not break other projects.
- I can publish patches if bugs are found, and bump the repositories affected.
- I want an easy way to develop and bugfix components, would be nice to see changes in storybook then publish them to registry when done.

Anyone done anything like this? Or have some tips & tricks to get a good workflow?

**Comment by Marklb** [comment]
This seems like something you would handle independent of Storybook, unless I don't understand the question or need more information on your goal.

Your stories are ideally maintained along with your components, in the same repo, but it builds into it's own application that you could host, like any other website. So, your Storybook application would most likely not be in an NPM registry, but you could have a link to the hosted Storybook application in your NPM package's `package.json` that lets projects depending on that package to have it composed into their Storybook application.

There is potentially a reason to publish stories to a registry, in your package, for [Storybook Composition](https://storybook.js.org/docs/sharing/package-composition), but I think the only supported solution for that is to reference the package's hosted Storybook. *I could be wrong, but I think what I am actually thinking of was changed or not completed, yet.
